src/core/primitives.js

```diff
@@ -81,59 +81,47 @@ var Dict = (function DictClosure() {
     },
 
     // automatically dereferences Ref objects
-    get: function Dict_get(key1, key2, key3) {
-      var value;
-      var xref = this.xref, suppressEncryption = this.suppressEncryption;
-      if (typeof (value = this._map[key1]) !== 'undefined' ||
-          key1 in this._map || typeof key2 === 'undefined') {
-        return xref ? xref.fetchIfRef(value, suppressEncryption) : value;
+    get(key1, key2, key3) {
+      let value = this._map[key1];
+      if (value === undefined && !(key1 in this._map) && key2 !== undefined) {
+        value = this._map[key2];
+        if (value === undefined && !(key2 in this._map) && key3 !== undefined) {
+          value = this._map[key3] || null;
+        }
       }
-      if (typeof (value = this._map[key2]) !== 'undefined' ||
-          key2 in this._map || typeof key3 === 'undefined') {
-        return xref ? xref.fetchIfRef(value, suppressEncryption) : value;
+      if (value instanceof Ref && this.xref) {
```

Comment:
Would it be better to (here and below) do the `this.xref` check first, given that it's probably cheaper than the `instanceof` check? These methods are usually called a lot, so maybe that's worth it.

Reply:
*It's possible that this is going too far into the territory of micro-optimizations, but:*
Statistically speaking most `Dict`s will have `this.xref` defined, whereas only some entries will contain references. By using this ordering, we can skip what most of the time amounts to redundant `this.xref` checks, but the difference is probably unmeasurable though...

Reply:
Right, I thought so, but thought it couldn't hurt to ask. Looks good in that case.